src/System.Management.Automation/engine/Modules/ModuleCmdletBase.cs

```diff
@@ -4768,6 +4811,62 @@ internal static Collection<string> GetResolvedPathCollection(string filePath, Ex
             return filePaths;
         }
 
+        internal PSSession GetWindowsPowerShellCompatRemotingSession()
+        {
+            PSSession result = null;
+            var commandInfo = new CmdletInfo("Get-PSSession", typeof(GetPSSessionCommand));
+            using var ps = System.Management.Automation.PowerShell.Create(RunspaceMode.CurrentRunspace);
+            ps.AddCommand(commandInfo);
+            ps.AddParameter("Name", WindowsPowerShellCompatRemotingSessionName);
+            var results = ps.Invoke<PSSession>();
```

Comment:
your suggestion is how it was done originally, like you say, because that's how it is used in many other places.
Then there was (imho valid) feedback that PS object should be under `using` statement, so I changed it.

Reply:
You can still create the PowerShell instance with `using` right? In the example above I didn't include `Create()` deliberately.